cvat-ui/src/components/annotation-page/standard-workspace/image-processing.tsx

```diff
@@ -0,0 +1,105 @@
+// Copyright (C) 2023 CVAT.ai Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import React, {
+    useCallback, useEffect, useRef,
+} from 'react';
+import { Canvas } from 'cvat-canvas-wrapper';
+import { useSelector, useDispatch } from 'react-redux';
+import { CombinedState } from 'reducers';
+import notification from 'antd/lib/notification';
+import { changeFrameAsync } from 'actions/annotation-actions';
+import { ImageFilter } from 'utils/image-processing';
+
+export default function ImageProcessingComponent(): JSX.Element | null {
+    const filtersRef = useRef<ImageFilter[]>([]);
+    const frame = useRef<number | null>(null);
+    useSelector((state: CombinedState) => {
+        filtersRef.current = state.settings.imageProcessing.filters;
+        frame.current = state.annotation.player.frame.number;
+    });
+
+    const canvasInstance = useSelector((state: CombinedState) => state.annotation.canvas.instance) as Canvas;
+    const states = useSelector((state: CombinedState) => state.annotation.annotations.states);
+    const curZOrder = useSelector((state: CombinedState) => state.annotation.annotations.zLayer.cur);
+    const frameData = useSelector((state: CombinedState) => state.annotation.player.frame.data);
+    const filters = useSelector((state: CombinedState) => state.settings.imageProcessing.filters);
+
+    const dispatch = useDispatch();
+
+    const getCanvasImageData = ():ImageData => {
+        const canvas: HTMLCanvasElement | null = window.document.getElementById('cvat_canvas_background') as
+        | HTMLCanvasElement
+        | null;
+        if (!canvas) {
+            throw new Error('Element #cvat_canvas_background was not found');
+        }
+        const { width, height } = canvas;
+        const context = canvas.getContext('2d');
+        if (!context) {
+            throw new Error('Canvas context is empty');
+        }
+        return context.getImageData(0, 0, width, height);
+    };
+
+    const runImageModifier = useCallback(async (): Promise<void> => {
+        try {
+            const currentFilters = filtersRef.current;
+            const currentFrame = frame.current;
+            if (currentFilters.length === 0 || currentFrame === null) {
+                return;
+            }
+            const imageIsNotProcessed = currentFilters.some((imageFilter: ImageFilter) => (
+                imageFilter.modifier.currentProcessedImage !== currentFrame
+            ));
+            if (imageIsNotProcessed) {
+                const imageData = getCanvasImageData();
```

Comment:
Here  we may use OffscreenCanvas to convert ImageBitmap returned by original data() method, to ImageData then apply filters and convert it back to imageBitmap with createImageBitmap.

We do not need to wait ``canvas.setup`` action and then read ImageData from HTMLCanvas.